src/OpenTracing/SpanOptions.php

```diff
@@ -43,7 +43,15 @@ public static function create(array $options)
 
                 case 'tags':
                     foreach ($value as $tag => $tagValue) {
-                        $spanOptions->tags[] = Tag::create($tag, $tagValue);
+                        if ($tag !== (string) $tag) {
+                            throw InvalidSpanOption::invalidTag($tag);
+                        }
+
+                        if (!is_scalar($tagValue) && method_exists($tagValue, '__toString')) {
```

Comment:
if you pass an array as `$tagValue` `is_scalar()` will return `false`, which means you will call `method_exists()` on an array (not an object), which is not valid